### DISCUSSION TITLE
@types/fabric createpngstream

### DISCUSSION THREAD

**Discussion by uoik** [DefinitelyTyped/DefinitelyTyped#56768] [Issues with a @types package]
The createpngstream method is not defined

**Comment by uoik** [comment]
Property 'createpngstream' does not exist on type 'Canvas'

**Comment by Fleker** [comment]
Looks like something that's Node-only:

> All of this simply creates Fabric canvas and renders text object onto it. Now, how to create an image of whatever is rendered on canvas? Using createPNGStream method available right on the canvas instance. createPNGStream returns Node's stream object, and can then be output into an image file using on('data'), and writing into the stream corresponding to an image file (fs.createWriteStream()).

> fabric.Canvas#createPNGStream is one of the methods specific to Node. Everything else works the same — you can still create objects as you usually would, add them on canvas, modify, render, and so on.

http://fabricjs.com/fabric-intro-part-4

It's technically not in the reference docs: http://fabricjs.com/docs/fabric.Canvas.html

[a comment by an automated account is not shown]

**Comment by CanRau** [comment]
Here's the related code https://github.com/fabricjs/fabric.js/blob/master/index.node.ts
